urlaf656038a0_Salihan04_MOBLIMA_tgz-pJ8-com_moblima_util_dataTestJ8/src/com/moblima/dataaccess/SerializeDB.java

```diff
@@ -0,0 +1,314 @@
+	
+	private List<Cineplex> cineplexes;
+	private List<Cinema> cinemas;
+	private List<Seat> seats;
+	private List<Movie> movies;
+	private List<ShowTime> showTimes;
+	private List<MovieGoer> movieGoers;
+	private List<MovieTicket> movieTickets;
+	private List<Transaction> transactions;
+	private List<Booking> bookings;
+	private List<PublicHoliday> publicHolidays;
+	private List<TicketPrice> ticketPrices;
+	private List<Staff> staffs;
+	
+	private Integer cineplexID;
+	private Integer seatID;
+	private Integer movieID;
+	private Integer showTimeID;
+	private Integer movieGoerID;
+	private Integer movieTicketID;
+	private Integer bookingID;
+	private Integer publicHolidayID;
+	private Integer staffID;
+	
+	private static String fileName = "serializeDB.dat";
+	private static SerializeDB instance;
+	
+	public static SerializeDB getInstance() {
+		if(instance == null) {
+			instance = new SerializeDB();
+		}
+		
+		return instance;
+	}
+	
+	public void saveData() {
+		try {
+			FileOutputStream fout = new FileOutputStream(fileName);
+			ObjectOutputStream oos = new ObjectOutputStream(fout);
+			
+			oos.writeObject(cineplexes);
+			oos.writeObject(cinemas);
+			oos.writeObject(seats);
+			oos.writeObject(movies);
+			oos.writeObject(showTimes);
+			oos.writeObject(movieGoers);
+			oos.writeObject(movieTickets);
+			oos.writeObject(transactions);
+			oos.writeObject(bookings);
+			oos.writeObject(publicHolidays);
+			oos.writeObject(ticketPrices);
+			oos.writeObject(staffs);
+			
+			oos.writeObject(cineplexID);
+			oos.writeObject(seatID);
+			oos.writeObject(movieID);
+			oos.writeObject(showTimeID);
+			oos.writeObject(movieGoerID);
+			oos.writeObject(movieTicketID);
+			oos.writeObject(bookingID);
+			oos.writeObject(publicHolidayID);
+			oos.writeObject(staffID);
+			
+			oos.close();			
+			
+		} catch(Exception e){
+			e.printStackTrace();
+		}
+	}
+	
+	private void loadData() {
+		File f = new File(fileName);
+		if(!f.exists()) {
+			try {
+				f.createNewFile();
+			} catch (IOException e) {
+				// TODO Auto-generated catch block
+				e.printStackTrace();
+			}
+			
+			cineplexes = new ArrayList<Cineplex>();
+			cinemas = new ArrayList<Cinema>();
+			seats = new ArrayList<Seat>();
+			movies = new ArrayList<Movie>();
+			showTimes = new ArrayList<ShowTime>();
+			movieGoers = new ArrayList<MovieGoer>();
+			movieTickets = new ArrayList<MovieTicket>();
+			transactions = new ArrayList<Transaction>();
+			bookings = new ArrayList<Booking>();
+			publicHolidays = new ArrayList<PublicHoliday>();
+			ticketPrices = new ArrayList<TicketPrice>();
+			staffs = new ArrayList<Staff>();
+			
+			cineplexID = 1;
+			seatID = 1;
+			movieID = 1;
+			showTimeID = 1;
+			movieGoerID = 1;
+			movieTicketID = 1;
+			bookingID = 1;
+			publicHolidayID = 1;
+			staffID = 1;
+		}
+		else {
+			try {
+				FileInputStream fin = new FileInputStream(fileName);
+				if(fin.available() > 0) {
+					ObjectInputStream ois = new ObjectInputStream(fin);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code contains a resource that might not be closed properly. This can cause a resource leak that slows down or crashes your system.

**Fix**
Consider closing the following resource: *ois*. The resource is referenced at lines 132, 133, 134, 135, 136, 137, 138, 139, 140, 141, 142, 143, 144, 145, 147, 148, 149, 150, 151, 152, 153, 154, 155. The resource is closed at line 156. There are other execution paths that do not contain closure statements, for example, when ObjectInputStream.readObject throws an exception. Ensure *ois* is closed in a try-finally block or declared in a try-with-resources block.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).